lib/adagrams.rb

```diff
@@ -0,0 +1,158 @@
+# method that creates a hand of letters for the user
+def draw_letters
+
+  # hash that store letters and their quantities
+  letter_pool = {
+    A: 9,
+    B: 2,
+    C: 2,
+    D: 4,
+    E: 12,
+    F: 2,
+    G: 3,
+    H: 2,
+    I: 9,
+    J: 1,
+    K: 1,
+    L: 4,
+    M: 2,
+    N: 6,
+    O: 8,
+    P: 2,
+    Q: 1,
+    R: 6,
+    S: 4,
+    T: 6,
+    U: 4,
+    V: 2,
+    W: 2,
+    X: 1,
+    Y: 2,
+    Z: 1,
+  }
+
+  # array to hold each indvidual letter in the letter pool as a string
+  letter_pool_array = []
+
+  # for each key in the pool, add the key in the quantity of its value into the letter_pool_array array
+  letter_pool.each do |letter, quantity|
+    letter_pool_array.fill(letter.to_s, letter_pool_array.size, quantity)
+  end
+
+  # pulling a random sample of 10 letters from the letter_pool_array
+  letters_in_hand = letter_pool_array.sample(10)
+
+  return letters_in_hand
+end
+
+# method that checks a user's input against the available letters to make sure they are valid
+def uses_available_letters?(input, letters_in_hand)
+
+  # splits the input into an array of characters
+  split_input = input.chars
+
+  # pushes true/false values into an array based on if user input is valid
+  true_false = []
+  split_input.each do |letter|
```

Comment:
I would suggest that if you ever encounter a letter not in the hand, that you can then return false.

Good catch in checking the count of the letter in the hand and the count of the letter in the input